File: course_101/exercises/easy_8/fizzbuzz.rb
# Write a method that takes two arguments: the first is the starting number, and the second is the ending number.
# Print out all numbers between the two numbers, except if a number is divisible by 3, print "Fizz",
# if a number is divisible by 5, print "Buzz", and finally if a number is divisible by 3 and 5, print "FizzBuzz".

# solution 1
def fizzbuzz(start_num, end_num)
  start_num.upto(end_num) do |num|
    if num % 5 == 0 && num % 3 == 0
      puts "FizzBuzz"
    elsif num % 3 == 0 then puts "Fizz"
    elsif num % 5 == 0 then puts "Buzz"
    else puts num
    end
  end
end

# solution 2
def fizzbuzz_value(num)
  case
  when num % 5 == 0 && num % 3 == 0
    'FizzBuzz'
  when num % 3 == 0
    'Fizz'
  when num % 5 == 0
    'Buzz'
  else
    num
  end
end

def fizzbuzz(start_num, end_num)
  values = []
  start_num.upto(end_num) do |num|
    values << fizzbuzz_value(num)
  end
  puts values
end

fizzbuzz(1, 15)

File: course_js210_computational_thinking_and_javascript_programming/lesson_3_practice_problems_logic_and_flow_control/fizz_buzz.js
// Write a function that iterates over the integers from 1 to 100, inclusive
// For multiples of three, log "Fizz" to the console
// For multiples of five, log "Buzz"
// For numbers which are multiples of both three and five, log "FizzBuzz"
// For all other numbers, log the number.

function fizzbuzz() {
  for(let i = 1; i <= 100; i++) {
    let fizzBuzz = ''

    if (i % 3 === 0) fizzBuzz += 'Fizz';
    if (i % 5 === 0) fizzBuzz += 'Buzz';

    console.log(fizzBuzz || i);
  }
} 

fizzbuzz();

// console output
// 1
// 2
// Fizz
// 4
// Buzz
// Fizz
// 7
// 8
// Fizz
// Buzz
// 11
// Fizz
// 13
// 14
// FizzBuzz
// 16
// 17
// Fizz
// ...